sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -136,13 +136,13 @@ object GpuHashJoin extends Arm {
 
   /** Determine if this type of join supports using the right side of the join as the build side. */
   def canBuildRight(joinType: JoinType): Boolean = joinType match {
-    case _: InnerLike | LeftOuter | LeftSemi | LeftAnti | _: ExistenceJoin => true
+    case _: InnerLike | LeftOuter | LeftSemi | LeftAnti | FullOuter | _: ExistenceJoin => true
     case _ => false
   }
 
   /** Determine if this type of join supports using the left side of the join as the build side. */
   def canBuildLeft(joinType: JoinType): Boolean = joinType match {
-    case _: InnerLike | RightOuter | FullOuter => true
```

Comment:
Agree (as per my comment above).  If building left, I'm thinking I'll need to use Right Outer join instead of Left Outer Join?

Reply:
To be clear here, this is not going to make any difference until we actually fix how we pick the build side for a sort merge join translation.

```
  val buildSide: GpuBuildSide = if (GpuHashJoin.canBuildRight(join.joinType)) {
    GpuBuildRight
  } else if (GpuHashJoin.canBuildLeft(join.joinType)) {
    GpuBuildLeft
  } else {
    throw new IllegalStateException(s"Cannot build either side for ${join.joinType} join")
  }
```

Perhaps there are cases if we enable hash joins in Spark that it might show up, but unless we can force this to happen in testing we are going to have one code path that is not tested.

Reply:
The logic for `sortMergeJoin` favors `GpuBuildRight` without looking at the join side from spark.  But I think in most other cases we use `getGpuBuildSide` and then match on that to check if we can build on that side.  So I think I do need to handle the buildLeft case as well.

Reply:
I have updated to support build left/right, but I didn't change this sortMergeJoin buildside code.  Should I go ahead and do that as part of this PR, or file a followup?